MealFridge/Models/Repositories/Repository.cs

```diff
@@ -18,14 +17,15 @@ public Repository(DbContext ctx)
             _context = ctx;
             _dbSet = _context.Set<TEntity>();
         }
+
         public virtual async Task<TEntity> AddOrUpdateAsync(TEntity entity)
         {
             if (entity == null)
             {
                 throw new ArgumentNullException("Entity must not be null to add or update");
             }
-            _context.Update(entity);
-            await _context.SaveChangesAsync();
+            _context.Add(entity);
```

Comment:
This shouldn't update entity then, but rather cause a db collision, so we'll need to double check it and see what it is doing.